hwbench/bench/monitoring.py

```diff
@@ -191,28 +185,19 @@ def __monitor(self, precision_s: int, frequency: int, duration_s: int) -> Monito
         end_of_run_ns = start_monitoring_ns + (duration_s * 1e9)
         loops_done = 0
         compact_count = 0
-        monitoring_start_adjusted = False
 
         def next_iter_ns() -> float:
             # When does the next iteration must starts ?
             return start_monitoring_ns + ((loops_done + 1) * precision_s) * 1e9
 
         # monitor_loop
         while True:
-            start_time_loop_ns = time.monotonic_ns()
-
-            # Get and parse the most recent turbostat sample from the buffer
+            # Trigger turbostat sample collection at the start of the monitoring loop
+            # This allows turbostat to collect metrics while we perform BMC/PDU monitoring
             if self.turbostat:
-                sample_timestamp = self.turbostat.get_and_parse_sample(precision_s)
+                self.turbostat.trigger_sample()
 
-                # On the first sample, synchronize our monitoring loop timing
-                if not monitoring_start_adjusted and sample_timestamp:
-                    start_monitoring_ns = self.turbostat.calculate_sync_offset(sample_timestamp, start_monitoring_ns)
-                    end_of_run_ns = start_monitoring_ns + (duration_s * 1e9)
-                    monitoring_start_adjusted = True
-
-                # Let's monitor the time spent at retrieving the CPU metrics, in milliseconds
-                self.metrics.contexts.Monitor.CPU["Polling"].add((time.monotonic_ns() - start_time_loop_ns) * 1e-6)
+            start_time_loop_ns = time.monotonic_ns()
```

Comment:
Shouldn't it be done before the trigger_sample() ?
Could you rename it to include turbostat in the name to make it more explicit ?